src/Common/TimerDescriptor.cpp

```diff
@@ -34,7 +34,8 @@ TimerDescriptor::TimerDescriptor(TimerDescriptor && other) noexcept : timer_fd(o
 
 TimerDescriptor & TimerDescriptor::operator=(DB::TimerDescriptor && other) noexcept
 {
-    std::swap(timer_fd, other.timer_fd);
+    timer_fd = other.timer_fd;
```

Comment:
i came across this code looking into this issue: https://github.com/ClickHouse/ClickHouse/issues/37686
not sure it causes any problems, but seems not quite right to me

Reply:
IIUC the original code is correct. It's an `assignment` which means something should clean up the old `timer_fd`. Using `std::swap` we rely `other`'s d'tor to do that. It's also a `move` which means nobody cares what state `other` is in afterwards. You may move from it. You may copy from it. You may swap into it, etc.

Reply:
yes, you're right. and now we will leak this->timer_fd because there will be no TimerDescriptor that will call close on it.